src/types/redis_tdigest.cc

```diff
@@ -161,6 +161,8 @@ rocksdb::Status TDigest::Add(engine::Context& ctx, const Slice& digest_name, con
 
   metadata.total_observations += inputs.size();
   metadata.total_weight += inputs.size();
+  metadata.maximum = std::max(metadata.maximum, *std::max_element(inputs.cbegin(), inputs.cend()));
```

Comment:
Hi @wyxxxcat ,

The `metadata.minimum` and `metadata.maximum` should represent the centroids min and max.
It will be updated in [merge action](https://github.com/apache/kvrocks/blob/34007db3a5a74e7c773209a16c6d0a5cc09ef884/src/types/tdigest.cc#L238) and updated by [dumping after merging](https://github.com/apache/kvrocks/blob/34007db3a5a74e7c773209a16c6d0a5cc09ef884/src/types/tdigest.cc#L393).

Maybe we could just compute inplace or create a new property in metadata or do a merge to keep the conaistency of metadata and centroids. 😊

Reply:
Hi @mapleFU ,

Thanks for your update! 😊

Currently the min and max are updated after Merge with `DumpCentroids()`.

But `Add` will not always trigger the Merge action until the buffer is full.

So, if we try to update it in `Add` maybe we'd better remove the [updating of min&max after `DumpCentroids()`](https://github.com/apache/kvrocks/blob/34007db3a5a74e7c773209a16c6d0a5cc09ef884/src/types/redis_tdigest.cc#L282-L283).

Since when we try to merge, the buffer will always be locked, it is safe to update the value either in `Add` or `Merge`.

Best Regards,
Edward

Reply:
Hi @mapleFU ,

I agree on your view. 😊
Frankly this probably just because I implement the `Merge` function ahead of `Add`. 😄

So, I think move it to the `Add` function is reasonable. But we'd better remove the updating after Merge if we decide to maintain it in `Add`.

Update it in two places may not be a good practice.

Best Regards,
Edward

Reply:
Hmm so I'm wondering if we should submit a PR to make these changes before this PR to be merged?

Reply:
Hi @PragmaTwice ,

Got it! 
I'll create a patch today to refactor the min max updating logic to `Add`.😊

Best Regards,
Edward